python/pyspark/sql/functions.py

```diff
@@ -798,7 +798,7 @@ def factorial(col):
 # ---------------  Window functions ------------------------
 
 @since(1.4)
-def lag(col, count=1, default=None):
+def lag(col, offset=1, default=None):
```

Comment:
Let's deprecate it first.

```python
from pyspark import _NoValue

def lag(col, offset=1, default=None, count=_NoValue):
    if count is not _NoValue:
        warnings.warn("'count' is deprecated in 3.0. Use 'offset' instead.", DeprecationWarning)
        offset = count
```

Reply:
Yeah that would be most conservative; I think we could fix it in 3.0 as a bug without this, but I'm OK with it too. If, weirdly, someone sets both, the old value would override the new, but maybe that's fine as corner case behavior.

Reply:
Ah, then, we can:

```python
from pyspark import _NoValue

def lag(col, offset=_NoValue, default=None, count=_NoValue):
    value = 1
    if count is not _NoValue:
        warnings.warn("'count' is deprecated in 3.0. Use 'offset' instead.", DeprecationWarning)
        value = count
    offset = offset if offset is not _NoValue else value
```

IIRC, there's few more cases like this. I don't super feel strongly about it too but I just wanted to be a bit more conservative to avoid to make this as an example.

Reply:
Hm ...  but this case `count` doesn't really makes sense. Let me just push this and probably we could consider the deprecation way above later ..